src/anndata/_core/index.py

```diff
@@ -30,11 +29,9 @@ def _normalize_indices(
         if len(index) > 2:
             # only one remaining ellipsis
             num_ellipsis = sum(i is Ellipsis for i in index)
-            if num_ellipsis == 1:
-                index = tuple(i for i in index if i is not Ellipsis)
-            elif num_ellipsis > 1:
+            if num_ellipsis > 1:
                 raise IndexError("an index can only have a single ellipsis ('...')")
-            if len(index) > 2:
+            if num_ellipsis == 0:
                 raise ValueError("AnnData can only be sliced in rows and columns.")
```

Comment:
seems like you do this check a second time in `unpack_index`.

if there is a reason, maybe extract it as a helper and add a comment explaining why you need to do it before calling `unpack_index`

Reply:
I checked, all this can just be

```suggestion
        elif isinstance(index, tuple):
            index = tuple(i.array if isinstance(i, pd.Series) else i for i in index)
```